src/components/WheelSelect/wheelPicker.js

```diff
@@ -0,0 +1,159 @@
+    const height = this.props.height || 40; // required to calculate which item should be selected on scroll
+    const dataLength = this.props.data.length;
+    const updateSelection = this.props.updateSelection;
+    const multiSelect = this.props.multiSelect || false;
+
+    //if there is already a timeout in process cancel it
+    if (this._scrollTimer) clearTimeout(this._scrollTimer);
+
+    let scroll = e.srcElement.scrollTop; // scroll value
+
+    // itemInSelectorArea height of area available to scroll / height of individual item
+    let itemInSelectorArea = parseInt((scroll + height / 2) / height);
+
+    if (itemInSelectorArea < dataLength) {
+      document
+        .getElementById(`wheel-picker__scroll-item--${itemInSelectorArea}`)
+        .classList.add('wheel-picker__selected-time');
+
+      for (let i = 0; i < dataLength; i++) {
+        if (i !== itemInSelectorArea) {
+          document
+            .getElementById(`wheel-picker__scroll-item--${i}`)
+            .classList.remove('wheel-picker__selected-time');
+        }
+      }
+    }
+
+    function finishedScrolling(selectorAreaHeight, id, updateSelection) {
+      updateSelection(itemInSelectorArea);
+      let fix = document.getElementById(id);
+      let y = itemInSelectorArea * selectorAreaHeight - 1;
+
+      if (y === -1) {
+        y = 0;
+      } else if (y >= (dataLength - 1) * height) {
+        y = (dataLength - 1) * height;
+      }
+
+      fix.scroll({
+        top: y,
+        behavior: 'smooth',
+      });
+    }
+
+    this._scrollTimer = setTimeout(
+      () => finishedScrolling(height, this.scrollerId, updateSelection),
+      200
+    );
+
+    ApplyStyle(this.scrollerId, itemInSelectorArea, dataLength, multiSelect);
+  }
+
+  itemClick(e, height) {
+    const m = e.target.id.split('--')[1];
+    document
+      .getElementById(this.scrollerId)
+      .scroll({ top: m * height - 1, behavior: 'smooth' });
+  }
+
+  renderListItems() {
+    const height = this.props.height || 40;
+    const multiSelect = this.props.multiSelect || false;
+
+    return this.props.data.map((item, index) => {
+      return (
+        <div
+          key={ index }
+          className="wheel-picker__scroll-item-container"
+          style={{ minHeight: height + 'px', maxHeight: height + 'px' }}
+        >
+          <div
+            role={ item.value }
+            key={ index }
+            id={ `wheel-picker__scroll-item--${index}` }
+            className={
+              multiSelect
+                ? 'wheel-picker__scroll-item wheel-picker__scroll-item--multiple'
+                : 'wheel-picker__scroll-item'
+            }
+            style={{
+              minHeight: height + 'px',
+              maxHeight: height + 'px',
+            }}
+            onClick={ (e) => this.itemClick(e, height) }
+            onKeyDown={ (e) => this.itemClick(e, height) }
+          >
+            { item.display }
+          </div>
+          { multiSelect && (
+            <div
```

Comment:
@rafayet-monon We can click outside even that we aren't on focus the item?🤔
![React_App](https://user-images.githubusercontent.com/749672/98900605-b43d6d00-24e4-11eb-8a46-49412efbb661.png)

Reply:
Yes, That shouldn't be the case? IMO, it's okay to be clickable even if it's not in focus. What do you think?

Reply:
@rafayet-monon I think it shouldn't clickable as it isn't in the highlight area 🤔

Reply:
Fixed in [c5b28df](https://github.com/rafayet-monon/surveyor/pull/19/commits/c5b28df9ef4447d2af72a49d3a350baeafea488e)